src/sort-filter-bar/sort-filter-bar.ts

```diff
@@ -341,46 +390,148 @@ export class SortFilterBar
+        displayCaret
+        closeOnSelect
+        includeSelectedOption
+        .openViaButton=${false}
+        .options=${options.dropdownOptions}
+        .selectedOption=${options.selectedOption}
+        @optionSelected=${options.onOptionSelected ?? nothing}
+        @click=${options.onDropdownClick ?? nothing}
       >
-        ${Object.keys(SortField).map(
-          field => html`
-            <option value="${field}" ?selected=${this.selectedSort === field}>
-              ${SortFieldDisplayName[field as SortField]}
-            </option>
-          `
-        )}
-      </select>
+        <span
+          class="dropdown-label"
+          slot="dropdown-label"
+          @click=${options.onLabelInteraction ?? nothing}
+          @keydown=${options.onLabelInteraction
+            ? (e: KeyboardEvent) => {
+                if (e.key === 'Enter' || e.key === ' ')
+                  options.onLabelInteraction?.();
+              }
+            : nothing}
+        >
+          ${options.displayName}
+        </span>
+      </ia-dropdown>
     `;
   }
 
-  private mobileSortChanged(e: Event) {
-    const target = e.target as HTMLSelectElement;
-    const sortField = target.value as SortField;
+  /** Generates a single dropdown option object for the given sort field */
+  private getDropdownOption(sortField: SortField): optionInterface {
+    return {
+      id: sortField,
+      selectedHandler: () => {
+        this.selectDropdownSortField(sortField);
+      },
+      label: html`
+        <span class="dropdown-option-label">
+          ${SortFieldDisplayName[sortField]}
+        </span>
+      `,
+    };
+  }
+
+  /** Handler for when any sort dropdown option is selected */
+  private dropdownOptionSelected(e: CustomEvent<{ option: optionInterface }>) {
+    this.dropdownBackdropVisible = false;
+    this.clearAlphaBarFilters();
+    this.setSelectedSort(e.detail.option.id as SortField);
+    this.emitTitleLetterChangedEvent();
+    this.emitCreatorLetterChangedEvent();
+  }
+
+  /** The template to render for the views dropdown */
+  private get viewsDropdownTemplate(): TemplateResult {
+    return this.getSortDropdown({
+      displayName: html`${this.viewSortField}`,
+      id: 'views-dropdown',
+      isSelected: () => this.viewOptionSelected,
+      dropdownOptions: [
+        this.getDropdownOption(SortField.weeklyview),
+        this.getDropdownOption(SortField.alltimeview),
+      ],
+      selectedOption: this.viewOptionSelected ? this.selectedSort : '',
+      onOptionSelected: this.dropdownOptionSelected,
+      onDropdownClick: () => {
+        this.dateDropdown.open = false;
+        this.dropdownBackdropVisible = this.viewsDropdown.open;
+        this.viewsDropdown.classList.toggle('open', this.viewsDropdown.open);
+      },
+      onLabelInteraction: () => {
+        if (!this.viewsDropdown.open && !this.viewOptionSelected)
+          this.setSelectedSort(SortField.weeklyview);
```

Comment:
QUESTION: Do we generally allow one-line `if` bodies with no braces? On two separate lines? (Same question applies below.)

Reply:
Thanks, I've added braces around this and the similar occurrences below.